Userland/Libraries/LibJIT/Assembler.h

```diff
@@ -140,8 +140,13 @@ struct Assembler {
                 | ((to_underlying(src.reg) >= 8) ? 1 << 2 : 0)
                 | ((to_underlying(dst.reg) >= 8) ? 1 << 0 : 0));
             emit8(0x89);
-            emit8(0x80 | (encode_reg(src.reg) << 3) | encode_reg(dst.reg));
-            emit32(dst.offset_or_immediate);
+            if (dst.offset_or_immediate <= 127) {
```

Comment:
Iff the offset (or as the spec calls it `displacement`) is `0` you can elide the immediate altogether
(Vol2 Table2.2 the `mod` bits would then be `0b00` instead of `0b01`/`0b10`)